api/kv_test.go

```diff
@@ -319,6 +319,22 @@ func TestExtractCustomMetadata(t *testing.T) {
 			},
 			expected: map[string]interface{}{"org": "eng"},
 		},
+		{
+			name: "a read response with no custom metadata",
```

Comment:
I think these will work great. I wonder if it's also worth adding a test case for a read and write response that has the `custom_metadata` field but where it's just an empty map. Just to make sure this function works when there _is_ a custom_metadata field but the custom metadata that was in there before got deleted (cuz I think it would end up as an empty map in that case, not nil. Not 100% sure but yeah might as well test?)

Reply:
Good call. I can add that test as well.